staging/src/k8s.io/apiserver/pkg/endpoints/filters/authentication.go

```diff
@@ -58,8 +60,19 @@ var (
 
 	authenticatedAttemptsCounter = metrics.NewCounterVec(
 		&metrics.CounterOpts{
-			Name: "authentication_attempts",
-			Help: "Counter of authenticated attempts.",
+			Name:           "authentication_attempts",
+			Help:           "Counter of authenticated attempts.",
+			StabilityLevel: metrics.ALPHA,
+		},
+		[]string{"result"},
+	)
+
+	authenticationLatency = metrics.NewHistogramVec(
+		&metrics.HistogramOpts{
+			Name:           "authentication_duration_seconds",
+			Help:           "Authentication duration in seconds broken out by result.",
+			Buckets:        prometheus.ExponentialBuckets(0.001, 2, 10),
```

Comment:
This means the largest bucket is only just over a second right?  I can see webhook tail latency being much higher.  15 steps would give you over 30 seconds which is generally the timeout for a request.  Granted I forget what we set the timeout for webhooks to.

Reply:
Thanks for your information. 
I can't agree more.